app/models/sms/incoming.rb

```diff
@@ -21,4 +18,12 @@ def recipient_numbers
   def recipient_hashes(options = {})
     [user: User::SITE, phone: to]
   end
+
+  private
+
+  def set_user
+    matching_users = User.by_phone(from).active
+    self.user ||= matching_users.first if matching_users.count == 1
+    true
+  end
```

Comment:
Do we need this? I thought when I looked that the only place this is needed is in sms receive, so we could remove it. If it is still needed, could you please add a comment explaining why it's in two places.